src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,195 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.commands.ByeCommand;
+import duke.commands.Command;
+import duke.commands.DeadlineCommand;
+import duke.commands.DeleteCommand;
+import duke.commands.DoneCommand;
+import duke.commands.EventCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+import duke.commands.SaveCommand;
+import duke.commands.TodoCommand;
+import duke.commands.UpdateCommand;
+import duke.exception.DukeException;
+import duke.tasks.DeadlineTask;
+import duke.tasks.EventTask;
+import duke.tasks.Task;
+import duke.tasks.TodoTask;
+
+/**
+ * Class that is responsible for parsing raw inputs and translating them to the correct Command or Task.
+ */
+public class Parser {
+
+    private static final int SINGLE_INPUT_COMMAND_SIZE = 1;
+    private static final int DOUBLE_INPUT_COMMAND_SIZE = 2;
+
+
+    /**
+     * Parses user input.
+     *
+     * @param input The user input.
+     * @return The Command subclass that correctly represents the user intent.
+     * @throws DukeException If input is unexpected or exceptional.
+     */
+    public static Command parseInput(String input) throws DukeException {
+        validateNotEmpty(input);
+
+        String[] segments = input.split(" ", 2);
+        String cmd = segments[0].trim();
+
+        // CHECK FOR NON-TASKS-RELATED COMMANDS
+        try {
+            switch (cmd) {
+            case "bye":
+                return new ByeCommand();
+            case "save":
+                return new SaveCommand();
+            case "list":
+                return new ListCommand();
+            case "done":
+                return new DoneCommand(Integer.parseInt(segments[1].trim()));
+            case "delete":
+                return new DeleteCommand(Integer.parseInt(segments[1].trim()));
+            case "find":
+                if (segments[1].length() == 0) {
+                    throw new IndexOutOfBoundsException();
+                }
+                return new FindCommand(segments[1].trim().split(" "));
+            case "update":
+                String[] updateSegments = segments[1].split(" ", 2);
+                if (updateSegments[1].trim().length() == 0) {
+                    throw new DukeException("Empty description is not valid");
+                }
+                return new UpdateCommand(
+                        Integer.parseInt(updateSegments[0].trim()),
+                        updateSegments[1].trim());
```

Comment:
Good use of switch statements to make the code neater.
You can consider handling the respective commands (i.e trim and split the string command for "find" and "update") in their respective classes to make this chunk of code neater.